mast_aladin_lite/table/mast_table.py

```diff
@@ -0,0 +1,172 @@
+
+import os
+import json
+from collections import defaultdict
+from traitlets import List, Unicode, Bool, Int, Any, observe
+from ipypopout import PopoutButton
+from ipyvuetify import VuetifyTemplate
+from ipywidgets.widgets import widget_serialization
+
+
+# locations for metadata from MissionMAST:
+missions = ['jwst', 'roman', 'hst']
+table_data_dir = os.path.join(os.path.dirname(__file__), 'data')
+unique_column_path = os.path.join(table_data_dir, 'unique_columns_per_mission.json')
+column_descriptions_path = os.path.join(table_data_dir, 'column_descriptions.json')
+
+# register loaded table widgets as they're initialized
+_table_widgets = dict()
+
+
+def serialize(table):
+    """
+    Convert an astropy table to a list of dictionaries
+    containing each column as a list of pure Python objects.
+    """
+    return [
+        {
+            k: v.tolist()
+            for k, v in dict(row).items()
+        } for row in table
+    ]
+
+
+class MastTable(VuetifyTemplate):
+    """
+    Table widget for observation queries from Mission MAST.
+    """
+    template_file = __file__, "mast_table.vue"
+
+    items = List().tag(sync=True)
+    headers_visible = List().tag(sync=True)
+    headers_avail = List().tag(sync=True)
+    show_if_empty = Bool(True).tag(sync=True)
+    show_rowselect = Bool(True).tag(sync=True)
+    selected_rows = List().tag(sync=True)
+    column_descriptions = List().tag(sync=True)
+    multiselect = Bool(True).tag(sync=True)
+    item_key = Unicode().tag(sync=True)
+    clear_btn_lbl = Unicode('Clear Table').tag(sync=True)
+    items_per_page = Int(5).tag(sync=True)
+    popout_button = Any().tag(sync=True, **widget_serialization)
+    show_tooltips = Bool(False).tag(sync=True)
+    menu_open = Bool(False).tag(sync=True)
+
+    table = None
+    row_select_callbacks = []
```

Comment:
question: since we are including this in the PR, we should also include a way to set it and an example of it being used. It would be cool to have aladin reorient to the selected row as an example